auction-app-frontend/src/components/SettingsTab/SettingsTab.jsx

```diff
@@ -1,36 +1,88 @@
-import { Button, Checkbox } from "src/components";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
 
-import { BUTTON_LABELS, POLICY_AND_COMMUNITY } from "src/constants";
+import { Button, ErrorComponent, LoadingComponent } from "src/components";
+
+import { BUTTON_LABELS, ROUTE_PATHS, BUTTON_VARIANTS } from "src/constants";
+import { useUser } from "src/store/UserContext";
+import { getUser, deactivateAccount } from "src/services/userService";
 
 import "./style.scss"
 
 const SettingsTab = () => {
-  return (
-    <div className="settings-tab-container">
-        <div className="box-container body-semibold">
-            <div className="box top-left">
-                <span className="box-heading">Policy and Community</span>
-                <span className="box-description">Receive updates on bids and seller's offers. Stay informed through:</span>
-                <div className="checkbox-container body-regular">
-                    { POLICY_AND_COMMUNITY.map((item) => (
-                        <Checkbox key={ item.id } label={ item.label } />
-                    )) }
+    const [user, setUser] = useState(null);
+    const [loading, setLoading] = useState(true);
+    const [updateLoading, setUpdateLoading] = useState(false);
+    const [error, setError] = useState(null);
+    const [warning, setWarning] = useState(null);
+
+    const { userId } = useUser();
+    const navigate = useNavigate();
+  
+    const fetchUserInformation = () => {
+      setLoading(true);
+  
+      getUser(userId)
+        .then((response) => {
+          setUser(response);
+        })
+        .catch((error) => {
+          setError(error);
+          setLoading(false);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  
+    const handleDeactivateAccount = () => {
+      setUpdateLoading(true);
+
+      deactivateAccount(userId)
+        .then(() => {
+          navigate(ROUTE_PATHS.HOME);
+
+          setUpdateLoading(false);
+          
+          localStorage.clear();
+        }).catch((error) => {
+          if ((error.response.data.trace).startsWith("com.example.auctionapp.exceptions.authentication.CannotDeactivateAccount")) {
```

Comment:
We shouldn't do it like this.
Throw exception on backend if this happens and just display error message.